### ISSUE TITLE
(doc) needs help on how to deal with batch data

### ISSUE THREAD

**Issue by qlibp** [minitorch/Module-2#5]
In `run_tensor.py`, how to implement the `forward` function to handle batch data? As the final result of `Network.forward` should return `Tensor` data, so the method in Module 1 using `for` and `list` is not a choice. For now, I merely figure out one possible solution: add one more dimension to the dataset. For dataset of shape `(50, 2)`, I will change it to `(50, 1, 2)` and then using the broadcasting mechanism to pass it through network.

Is there a better way to implement it?

**Comment by srush** [comment]
You are correct that this is the expected method to implement it  => View, then Broadcast, and then reduce. I will add some hints to make that more clear.

**Comment by srush** [comment]
I will add some more docs here, in the mean time, this diagram from the next section explains the main approach https://minitorch.github.io/matrixmult.html

**Comment by qlibp** [comment]
> You are correct that this is the expected method to implement it => View, then Broadcast, and then reduce. I will add some hints to make that more clear.

Thanks for the confirmation and then I will try to figure out what goes wrong with my code.

**Comment by qlibp** [comment]
Struggle for more than a week, still can't make it correct. The situation right now is that all tests are passed and the run_tensor.py run without error/warning. But I could not do the classification correctly. The parameters do not update at some point as the grads are small. Hope that you may have some thoughts on this and give a me hint.

Thx!

**Comment by qlibp** [comment]
I finally find out that

**Comment by qlibp** [comment]
I finally find out that the reason why my training procedure get stuck(derivative vanish at around epoch=60). I miss the relu activation in the Network forward function. I comment here in case someone come up with the same problem which I waste for about a week to find it out.

**Comment by srush** [comment]
Oh no! I didn't mean for that to be a hard part. I'll add more docs in the network class to describe the exact shape of the network. 

Did you eventually get it working? I see why that would be frustrating.

**Comment by qlibp** [comment]
Oh yeah, I finally make it work(maybe). It's still a little bit different from the result of Module 1.

<img width="1440" alt="截屏2020-09-13 下午11 18 20" src="https://user-images.githubusercontent.com/43082379/93021847-c18ac800-f617-11ea-9e2f-a29f953431ba.png">

**Comment by srush** [comment]
Looks great!
